### ISSUE TITLE
SC2223 causes incorrect behavior

### ISSUE THREAD

**Issue by kpedro88** [koalaman/shellcheck#3286]
#### For bugs with existing features

- Rule Id (if any, e.g. SC1000): SC2223
- My shellcheck version (`shellcheck --version` or "online"): 0.11.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

Consider the following:
```bash
#!/bin/bash

: "${TESTVAR1:=~/.bashrc}"

if [ -e "$TESTVAR1" ]; then
    echo "yay"
else
    echo "boo"
fi

: ${TESTVAR2:=~/.bashrc}

if [ -e "$TESTVAR2" ]; then
    echo "yay"
else
    echo "boo"
fi
```

This script (in various versions of Bash from 4.2.46 to 5.2.21, at least) prints:
```
boo
yay
```

Enabling `set -x` shows why:
```
+ : '~/.bashrc'
+ '[' -e '~/.bashrc' ']'
+ echo boo
boo
+ : /home/user/.bashrc
+ '[' -e /home/user/.bashrc ']'
+ echo yay
yay
```

#### Here's what shellcheck currently says:

```
Line 11:
: ${TESTVAR2:=~/.bashrc}
  ^-- [SC2223](https://www.shellcheck.net/wiki/SC2223) (info): This default assignment may cause DoS due to globbing. Quote it.
```

#### Here's what I wanted or expected to see:

ShellCheck flags the second version as violating SC2223, even though it gives the right answer. The first version passes SC2223, but gives the wrong answer, because the variable value is quoted as a literal rather than getting expanded as intended.

This pattern is a common use of shell parameter expansion to set a default value for a variable only if the user has not already set a value. Currently, I am disabling SC2223 globally for any affected script.

Even its [wiki page](https://www.shellcheck.net/wiki/SC2223) describes the concern as "largely theoretical", while the incorrect behavior it causes can be very simply demonstrated and clearly has negative practical impacts. This check should simply be removed. Alternatively, it should only trigger on values that include an asterisk.
